src/bitset.rs

```diff
@@ -22,7 +26,7 @@ pub const SHIFT3: usize = SHIFT2 + BITS;
 /// a certain component exists. It does not track the `Generation` of the
 /// entities that it contains.
 ///
-/// Note, a `BitSet` is limited by design to only 1,048,576 indices.
+/// Note, a `BitSet` is limited by design to only 1,048,576 indices for 32 bit and 16,777,216 indices for 64 bit architectures.
```

Comment:
It's really cool that you added comments about `BitSet`. But I prefer if it will be separate PR.